### PR TITLE
Masked select decomposition

### PR DESCRIPTION
**Pull request by jackzhxng**
Needed since ExecuTorch has no portable kernel for `aten.masked_select`.

### CODE REVIEW

**Comment by larryliu0820** [comment]
I think this is ok

**Review by larryliu0820 — APPROVED** [review]

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -430,6 +431,47 @@ def safe_softmax(self, dim, dtype=None):
     return torch.where(masked_rows, zeros, out)
 
 
+@register_decomposition(aten.masked_select)
```
Comment (Chillee) [inline torch/_decomp/decompositions.py:434]: Are there any tests for this?

**Review by Chillee — CHANGES_REQUESTED** [review]

**Review by angelayi — APPROVED** [review]
Looks good to me! It pretty much matches the [meta kernel](https://github.com/pytorch/pytorch/blob/504904c9c66fc89eada912b1b7d9fdf1c3e56b14/torch/_subclasses/fake_impls.py#L473-L511) too.

File: torch/_decomp/decompositions.py (hunk 1 above)
Comment (dvorjackz) [reply torch/_decomp/decompositions.py:434]: I don't think so, would you be able to point me to where the decomp tests are? I try adding one
